src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -214,40 +218,61 @@ private void IgnoreUnsupportProjectReference(PackageSpec project)
 
         #region NuGetProject
 
-        public override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
+        public async override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
         {
-            PackageReference[] installedPackages;
-
             var packageSpec = GetPackageSpec();
+
             if (packageSpec != null)
             {
-                installedPackages = GetPackageReferences(packageSpec);
-            }
-            else
-            {
-                installedPackages = new PackageReference[0];
+                return await GetPackageReferencesAsync(packageSpec);
             }
 
-            return Task.FromResult<IEnumerable<PackageReference>>(installedPackages);
+            return Array.Empty<PackageReference>();
         }
 
-        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        private async Task<IEnumerable<PackageReference>> GetPackageReferencesAsync(PackageSpec packageSpec)
```

Comment:
nitpick (no need to change): I don't like methods that return `IEnumerable`. It's understandable when the API is returning some streaming data, like from a HTTP API or databse. But when the data is already in memory, I see little benefit to returning `IEnumerable`. Particularly when the caller wants to know the number of items in the enumeration, then it needs to either enumerate the result multiple times (throws an exception in streaming scenarios, like the http or database examples), or it needs to call `ToList` or `ToArray`, but when the return type was downcast from a list or array, then it's a waste of CPU time and memory allocations.

It's unfortunately common in NuGet, but I would not recommend learning this practise, which I consider bad. An advantage is it allows the team to refactor the code and change the object being returned without breaking the API, but returning `IList` or `IReadOnlyList`, which are castable from both `List` and `Array` lets callers who care about count or indexers to be more efficient.

However, this method is private, and it's not in the NuGet SDK, so it's not a big deal. More of an FYI.